File: test_fatsecret.js
const https = require('https');

const CLIENT_ID = 'f7a036bce03d407082f0051ed0c8639a';
const CLIENT_SECRET = '7d1ca9b45a964c3095f04a0cdec04ee3';

function post(url, data, headers) {
    return new Promise((resolve, reject) => {
        const urlObj = new URL(url);
        const options = {
            hostname: urlObj.hostname,
            path: urlObj.pathname + urlObj.search,
            method: 'POST',
            headers: headers
        };

        const req = https.request(options, res => {
            let body = '';
            res.on('data', chunk => body += chunk);
            res.on('end', () => resolve({ statusCode: res.statusCode, body: body }));
        });

        req.on('error', reject);
        req.write(data);
        req.end();
    });
}

function get(url, headers) {
    return new Promise((resolve, reject) => {
        const urlObj = new URL(url);
        const options = {
            hostname: urlObj.hostname,
            path: urlObj.pathname + urlObj.search,
            method: 'GET',
            headers: headers
        };

        const req = https.request(options, res => {
            let body = '';
            res.on('data', chunk => body += chunk);
            res.on('end', () => resolve({ statusCode: res.statusCode, body: body }));
        });

        req.on('error', reject);
        req.end();
    });
}

async function test() {
    console.log("Testing FatSecret API...");

    // 1. Get Token
    const tokenUrl = 'https://oauth.fatsecret.com/connect/token';
    const auth = Buffer.from(`${CLIENT_ID}:${CLIENT_SECRET}`).toString('base64');
    const headers = {
        'Authorization': `Basic ${auth}`,
        'Content-Type': 'application/x-www-form-urlencoded'
    };
    const data = 'grant_type=client_credentials&scope=basic';

    try {
        console.log(`Requesting token from ${tokenUrl}...`);
        const tokenRes = await post(tokenUrl, data, headers);
        console.log(`Token Response Status: ${tokenRes.statusCode}`);
        console.log(`Token Response Body: ${tokenRes.body}`);

        if (tokenRes.statusCode !== 200) {
            console.log("Failed to get token.");
            return;
        }

        const token = JSON.parse(tokenRes.body).access_token;
        console.log("Token obtained successfully.");

        // 2. Search Food
        const searchUrl = `https://platform.fatsecret.com/rest/server.api?method=foods.search&search_expression=apple&format=json`;
        const searchHeaders = {
            'Authorization': `Bearer ${token}`
        };

        console.log(`Searching for 'apple' at ${searchUrl}...`);
        const searchRes = await get(searchUrl, searchHeaders);
        console.log(`Search Response Status: ${searchRes.statusCode}`);
        console.log(`Search Response Body: ${searchRes.body.substring(0, 500)}`);

    } catch (e) {
        console.error("Exception:", e);
    }
}

test();

File: test_fatsecret.py
import requests
import json
import base64

CLIENT_ID = 'f7a036bce03d407082f0051ed0c8639a'
CLIENT_SECRET = '7d1ca9b45a964c3095f04a0cdec04ee3'

def test_fatsecret():
    print("Testing FatSecret API...")
    
    # 1. Get Token
    token_url = 'https://oauth.fatsecret.com/connect/token'
    auth = requests.auth.HTTPBasicAuth(CLIENT_ID, CLIENT_SECRET)
    data = {'grant_type': 'client_credentials', 'scope': 'basic'}
    
    print(f"Requesting token from {token_url}...")
    try:
        response = requests.post(token_url, data=data, auth=auth)
        print(f"Token Response Status: {response.status_code}")
        print(f"Token Response Body: {response.text}")
        
        if response.status_code != 200:
            print("Failed to get token.")
            return

        token = response.json()['access_token']
        print("Token obtained successfully.")
        
        # 2. Search Food
        search_url = 'https://platform.fatsecret.com/rest/server.api'
        params = {
            'method': 'foods.search',
            'search_expression': 'apple',
            'format': 'json'
        }
        headers = {'Authorization': f'Bearer {token}'}
        
        print(f"Searching for 'apple' at {search_url}...")
        response = requests.get(search_url, params=params, headers=headers)
        print(f"Search Response Status: {response.status_code}")
        print(f"Search Response Body: {response.text[:500]}")
        
    except Exception as e:
        print(f"Exception: {e}")

if __name__ == "__main__":
    test_fatsecret()
